src/EventStore.Core/Services/Storage/EpochManager/EpochManager.cs

```diff
@@ -203,7 +203,8 @@ public class EpochManager : IEpochManager {
 			// and update EpochManager's state, by adjusting cache of records, epoch count and un-caching 
 			// excessive record, if present.
 			// If we are writing the very first epoch, last position will be -1.
-			var epoch = WriteEpochRecordWithRetry(_lastEpochNumber + 1, Guid.NewGuid(), _lastEpochPosition, _instanceId);
+			
+			var epoch = WriteEpochRecordWithRetry(epochNumber, Guid.NewGuid(), _lastEpochPosition, _instanceId);
```

Comment:
Should we add a pre-condition here that epochNumber > _lastEpochNumber and throw otherwise? (to make sure we catch any bugs)